subworkflows/local/utils_nfcore_fastqrepair_pipeline/main.nf

```diff
@@ -155,38 +157,58 @@ def validateInputSamplesheet(input) {
     def (metas, fastqs) = input[1..2]
 
     // Check that multiple runs of the same sample are of the same datatype i.e. single-end / paired-end
-    def endedness_ok = metas.collect{ it.single_end }.unique().size == 1
+    def endedness_ok = metas.collect{ meta -> meta.single_end }.unique().size == 1
     if (!endedness_ok) {
         error("Please check input samplesheet -> Multiple runs of a sample must be of the same datatype i.e. single-end or paired-end: ${metas[0].id}")
     }
 
     return [ metas[0], fastqs ]
 }
 
+//
+// Same fastq files are not allowed to be analyzed multiple times in the same run
+//
+def validateInputSamplesheet2(input) {
+    s = 0
+    input.size().map{
+        s = it
+
+        sunique = 0
+        input.toSet().size().map{
+            sunique = it
+
+            if(s != sunique){
+                error("\nPlease check input samplesheet -> Multiple runs of a fastq file are not allowed")
+            }
+        }
+    }
+}
+
 //
 // Generate methods description for MultiQC
 //
 def toolCitationText() {
-    // TODO nf-core: Optionally add in-text citation tools to this list.
     // Can use ternary operators to dynamically construct based conditions, e.g. params["run_xyz"] ? "Tool (Foo et al. 2023)" : "",
-    // Uncomment function in methodsDescriptionText to render in MultiQC report
     def citation_text = [
             "Tools used in the workflow included:",
-            "FastQC (Andrews 2010),",
-            "MultiQC (Ewels et al. 2016)",
-            "."
+            "The gzip Recovery Toolkit (https://www.urbanophile.com/arenn/hacking/gzrt/gzrt.html),",
+            "FastqWiper (https://github.com/mazzalab/fastqwiper),",
+            "Trimmomatic (Bolger et al. 2014),",
+            "BBMap (https://sourceforge.net/projects/bbmap/),",
+            "FastQC (Andrews 2010)"
```

Comment:
```suggestion
            "FastQC (Andrews 2010)."
```

It;s grea you've added this! But the effort was for nothing if you don't have the MultiQC because that's where this gets displayed which would be sad!